lib/crypto/sr25519/sr25519.go

```diff
@@ -43,6 +44,20 @@ type PrivateKey struct {
 	key *sr25519.SecretKey
 }
 
+// SignVerify verify signature given pubkey and msg
+func SignVerify(pubkey, sig, msg []byte) (bool, error) {
+	pubKey, err := NewPublicKey(pubkey)
+	if err != nil {
+		return false, fmt.Errorf("failed to fetch sr25519 public key: %s", err)
```

Comment:
I know this is not new code, but let's patch it up whilst we're at it :wink: 

Using `%w` to wrap the error

```suggestion
		return false, fmt.Errorf("failed to fetch sr25519 public key: %w", err)
```